tree/dataframe/src/RDFNodes.cxx

```diff
@@ -483,7 +483,7 @@ void RLoopManager::RunTreeProcessorMT()
 void RLoopManager::RunTreeReader()
 {
    TTreeReader r(fTree.get());
-   if (0 == fTree->GetEntriesFast())
+   if (0 == fTree->GetEntries())
```

Comment:
If fTree is a TChain, this would provoked open all the files.  Maybe
```
   fTree->GetTree() == nullptr && fTree->GetTree()->GetEntries() == 0
```

Reply:
what if the chain has 2 trees and one invokes: "fTree->GetTree()"?

Reply:
The objective we need to achieve is:
1) Be as fast as possible
2) Be able to skip corrupted files
In case the chain contains a corrupted file, see jira item, then the test for the validity of fTree will be of no use.

Reply:
?> what if the chain has 2 trees and one invokes: "fTree->GetTree()"?

It returns the 'current' TTree set by the last call to LoadTree.

Reply:
> Be as fast as possible

Then we never call TChain::GetEntries .... :)

Reply:
@dpiparo iiuc the problem is that `GetEntriesFast` does not check file validity and therefore we crash badly during the event loop while `GetEntries` opens files and checks their validity so zombie files are skipped automatically while looping?

Reply:
> In case the chain contains a corrupted file, see jira item, then the test for the validity of fTree will be of no use

In those case fTree->GetTree() ought to be nullptr.   (See code in TChain::LoadTree).

Reply:
I don't understand. What is, in code, the lightweight way to check the validity of the trees lumped together in a chain?

Reply:
There is no light weight way of checking them all before hand (because this requires opening each file and that is by nature 'slow').   What can be done is checking the value of TTree::GetTree after the call to LoadTree.

Reply:
Ok, but what shall I do in case I have a TChain with multiple files? What is the algorithm to check all of them?

Reply:
@pcanal if I understand correctly, the only way to not crash when running over a TChain in which some files are "bad" is to check `fChain->GetTree() != nullptr` after every call to `TChain::LoadTree` (i.e. after every file switch).

But afaik there is no lightweight way to know for what entries the `TChain` calls `LoadTree`, so we will have to call `fChain->GetTree() != nullptr` _at every entry_..?

Reply:
We could lower this into `TTreerReader`; it knows already when it's switching files. Would that help?